pkg/apis/garden/validation/validation.go

```diff
@@ -1852,6 +1892,76 @@ func ValidateWorker(worker garden.Worker, fldPath *field.Path) field.ErrorList {
 	if len(worker.Taints) > 0 {
 		allErrs = append(allErrs, validateTaints(worker.Taints, fldPath.Child("taints"))...)
 	}
+	if worker.Kubelet != nil {
+		allErrs = append(allErrs, ValidateKubeletConfig(*worker.Kubelet, fldPath.Child("kubelet"))...)
+	}
+
+	return allErrs
+}
+
+// ValidateWorker validates the worker object.
+func ValidateKubeletConfig(kubeletConfig garden.KubeletConfig, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if kubeletConfig.MaxPods != nil {
+		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*kubeletConfig.MaxPods), fldPath.Child("maxPods"))...)
+		// PodCIDR is /24
+		if *kubeletConfig.MaxPods > 256 {
```

Comment:
```suggestion
		if *kubeletConfig.MaxPods > 254 {
```

I guess the first and the last IP are reserved?

Reply:
Any reason you expect the `podCIDR` to be `/24`?

We recently changed this: https://github.com/gardener/gardener/pull/1280

Reply:
Bummer, did not notice it. I will have to calculate it accordingly. 
Thanks!